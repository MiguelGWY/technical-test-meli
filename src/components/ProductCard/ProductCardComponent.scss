.container-product {
  margin-bottom: 10px;

  &__product-elements {
    display: flex;
    padding: 20px;
    border-radius: 4px;
    border: thin solid #eee;
  }

  &__img {
    padding-right: 20px;
  }

  &__img img {
    height: 150px;
    width: 150px;
    object-fit: contain;
  }

  &__detail {
    display: grid;
  }

  &__title p {
    color: #333;
    font-size: 20px;
    font-weight: 300;
  }

  &__price {
    font-size: 24px;
    color: #333;
  }

  &__condition {
    background: #ffffff;
    color: #00a650;
    padding-left: 0;
    padding-right: 0;
  }
}
