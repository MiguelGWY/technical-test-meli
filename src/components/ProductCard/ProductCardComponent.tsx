import { formatCondition } from "../../utils/formattedCondition";
import { formattedPrice } from "../../utils/formattedPrice";
import "./ProductCardComponent.scss";

export const ProductCardComponent = ({
  title,
  picture,
  price,
  condition,
}: Item) => {
  const nameSpace = "container-product";

  return (
    <section className={nameSpace}>
      <div className={`${nameSpace}__product-elements`}>
        <div className={`${nameSpace}__img`}>
          <img src={picture} alt="Producto" />
        </div>
        <div className={`${nameSpace}__detail`}>
          <div className={`${nameSpace}__title`}>
            <p>{title}</p>
          </div>
          <div className={`${nameSpace}__price`}>{formattedPrice(price)}</div>
          <div className={`${nameSpace}__condition`}>
            {formatCondition(condition)}
          </div>
        </div>
      </div>
    </section>
  );
};
