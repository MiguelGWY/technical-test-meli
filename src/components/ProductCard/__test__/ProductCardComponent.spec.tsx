import { render } from "@testing-library/react";
import { ProductCardComponent } from "../ProductCardComponent";
import "@testing-library/jest-dom";

describe("ProductCardComponent", () => {
  const itemMock = {
    id: "1",
    title: "test product",
    price: {
      amount: 1234,
      decimals: 0,
      currency: "ARG",
    },
    thumbnail: "www.ddd.com",
    picture: "www.ddd.com",
    condition: "new",
    free_shipping: false,
    sold_quantity: 304,
    description: "Description Item",
    categories: [""],
    category_id: "112DADSAD",
    currency_id: "SDDDWEEEF",
  };

  it("should render ProductCardComponent", () => {
    const { container } = render(<ProductCardComponent {...itemMock} />);
    expect(container).toMatchSnapshot();
  });

  it("should render the product title, image, price, and condition", () => {
    const { getByText } = render(
      <ProductCardComponent {...itemMock} />
    );

    expect(getByText("test product")).toBeInTheDocument();
    expect(getByText("$ 1234")).toBeInTheDocument();
    expect(getByText("Nuevo")).toBeInTheDocument();
  });
});
