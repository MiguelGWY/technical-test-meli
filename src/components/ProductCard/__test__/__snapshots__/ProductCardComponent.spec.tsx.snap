// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardComponent should render ProductCardComponent 1`] = `
<div>
  <section
    class="container-product"
  >
    <div
      class="container-product__product-elements"
    >
      <div
        class="container-product__img"
      >
        <img
          alt="Producto"
          src="www.ddd.com"
        />
      </div>
      <div
        class="container-product__detail"
      >
        <div
          class="container-product__title"
        >
          <p>
            test product
          </p>
        </div>
        <div
          class="container-product__price"
        >
          $ 1234
        </div>
        <div
          class="container-product__condition"
        >
          Nuevo
        </div>
      </div>
    </div>
  </section>
</div>
`;
